### PR TITLE
refactor: Restructure src directory

### PR DESCRIPTION
**Pull request by jhermsmeier**
This restructures the `src/`, to compact it a bit and enable adding more files for the respective platforms without cluttering the directory or having to change the gypfile.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "build": "node-gyp build",
     "readme": "jsdoc2md --template docs/README.hbs index.js > README.md",
     "lint-js": "eslint *.js",
-    "lint-cpp": "cpplint src/*",
+    "lint-cpp": "cpplint src/*.* src/**/*.*",
```
Comment (jhermsmeier) [inline package.json]: Oh yeah, sure – mended! Didn't know that flag existed.

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "build": "node-gyp build",
     "readme": "jsdoc2md --template docs/README.hbs index.js > README.md",
     "lint-js": "eslint *.js",
-    "lint-cpp": "cpplint src/*",
+    "lint-cpp": "cpplint --recursive src/*",
```
Comment (jviotti) [inline package.json]: I'm afraid that will still fail, given that Windows will think `src/*` is literally a file.`cpplint --recursive src` works for me.

Comment (jhermsmeier) [inline package.json]: Really? Afaik singe asterisk wildcards work (i.e. `lib/*.js`), but maybe that was just a dream. Mended again :)

Comment (jviotti) [inline package.json]: Maybe its cpplint specific? I've used wildcards on Windows on npm scripts before, but this tool seems to take them literally for some reason.

**Pull request merged**